Exercise_4.java

```diff
@@ -5,15 +5,64 @@
     // Second subarray is arr[m+1..r] 
     void merge(int arr[], int l, int m, int r) 
     {  
-       //Your code here  
+       int t1 = m - l + 1; 
+        int t2 = r - m; 
+  
+       // Temp arrays to be merged
+        int temp1[] = new int [t1]; 
+        int temp2[] = new int [t2]; 
+  
+        for (int i=0; i<t1; i++) {
+            temp[i] = arr[l + i]; 

```

Comment:
You haven't declared temp anywhere in your code.
Modify this to temp1 and the one in line 19 to temp2.